File: qc-domain/src/main/java/com/eager/questioncloud/domain/coupon/CouponEntity.java
package com.eager.questioncloud.domain.coupon;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "coupon")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class CouponEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String code;

    @Column
    private String title;

    @Column
    @Enumerated(EnumType.STRING)
    private CouponType couponType;

    @Column
    private int value;

    @Column
    private int remainingCount;

    @Column
    private LocalDateTime endAt;

    @Builder
    public CouponEntity(Long id, String code, String title, CouponType couponType, int value, int remainingCount, LocalDateTime endAt) {
        this.id = id;
        this.code = code;
        this.title = title;
        this.couponType = couponType;
        this.value = value;
        this.remainingCount = remainingCount;
        this.endAt = endAt;
    }

    public Coupon toDomain() {
        return Coupon.builder()
            .id(id)
            .code(code)
            .title(title)
            .couponType(couponType)
            .value(value)
            .remainingCount(remainingCount)
            .endAt(endAt)
            .build();
    }

    public static CouponEntity from(Coupon coupon) {
        return CouponEntity.builder()
            .id(coupon.getId())
            .code(coupon.getCode())
            .title(coupon.getTitle())
            .couponType(coupon.getCouponType())
            .value(coupon.getValue())
            .remainingCount(coupon.getRemainingCount())
            .endAt(coupon.getEndAt())
            .build();
    }
}

File: qc-infra/qc-rdb/src/main/kotlin/com/eager/questioncloud/payment/entity/CouponEntity.kt
package com.eager.questioncloud.payment.entity

import com.eager.questioncloud.payment.domain.Coupon
import com.eager.questioncloud.payment.enums.CouponType
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "coupon")
class CouponEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Long = 0,
    @Column var code: String,
    @Column var title: String,
    @Column @Enumerated(EnumType.STRING) var couponType: CouponType,
    @Column var value: Int,
    @Column var remainingCount: Int,
    @Column var endAt: LocalDateTime
) {
    fun toDomain(): Coupon {
        return Coupon(id, code, title, couponType, value, remainingCount, endAt)
    }
    
    companion object {
        fun from(coupon: Coupon): CouponEntity {
            return CouponEntity(
                coupon.id,
                coupon.code,
                coupon.title,
                coupon.couponType,
                coupon.value,
                coupon.remainingCount,
                coupon.endAt
            )
        }
    }
}
